perf/ssl_ctx.c

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <openssl/ssl.h>
+#include "perflib/perflib.h"
+
+#define NUM_CTX_PER_RUN        100000
+
+int err = 0;
+
+OSSL_TIME *times;
+static char *certpath = NULL;
+static char *keypath = NULL;
+static char *storepath = NULL;
+
+static int threadcount;
+
+typedef enum {
+    OP_SERVER,
+    OP_CLIENT
+} op_mode;
+
+op_mode mode = OP_SERVER;
+
+static void do_create_ctx(size_t num)
+{
+    int ret = 1;
+    int i;
+    SSL_CTX *ctx = NULL;
+    OSSL_TIME start, end;
+
+    start = ossl_time_now();
+
+    for (i = 0; i < NUM_CTX_PER_RUN / threadcount; i++) {
+        ctx = SSL_CTX_new(mode == OP_SERVER ? TLS_server_method() :
+                                              TLS_client_method());
+        if (ctx == NULL)
+            goto out;
+        if (mode == OP_SERVER) {
+            if ((SSL_CTX_use_certificate_file(ctx, certpath,
+                                              SSL_FILETYPE_PEM) == 1) ||
+                (SSL_CTX_use_PrivateKey_file(ctx, keypath,
+                                             SSL_FILETYPE_PEM) == 1))
+                goto out;
+        } else {
+            if (SSL_CTX_load_verify_dir(ctx, storepath) != 1)
+                goto out;
+        }
+        SSL_CTX_free(ctx);
+        ctx == NULL;
+    }
+
+out:
+    end = ossl_time_now();
+    times[num] = ossl_time_subtract(end, start);
+
+    if (!ret)
+        err = 1;
+}
+
+static void usage(char *name)
+{
+    printf("usage\n");
+    printf("%s [-m <server|client>] [-c <cert>] [-k <key>] [-s <store>] <threadcount> \n", name);
+    printf("-m <server|client> - create a client or server method in context\n");
+    printf("-c <cert> - path to certificate for server context\n");
+    printf("-k <key> - path to key for server context\n");
+    printf("-s <store> - path to cert store for client context\n");
+}
+
+int main(int argc, char *argv[])
+{
+    OSSL_TIME duration, ttime;
+    uint64_t us;
+    double avcalltime;
+    int ret = EXIT_FAILURE;
+    int i;
+    int terse = 0;
+    int ch = 0;
+
+    while ((ch = getopt(argc, argv, "m:c:k:s:")) != -1) {
```

Comment:
`tesrs` variable is never set by `getopt()` loop, it remains set to 0. is that intentional?